### PR TITLE
RegExp should not be inlined while lifting

### PR DESCRIPTION
**Pull request by oicu0619**
related to #249

### CODE REVIEW

File: Sources/Fuzzilli/Lifting/JSExpressions.swift [hunk 1]
```diff
@@ -15,6 +15,7 @@
 // JavaScript expressions. See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Operator_Precedence
 public let Identifier           = ExpressionType(precedence: 20,                        inline: .always)
 public let Literal              = ExpressionType(precedence: 20,                        inline: .always)
+public let RegExpLiteral        = ExpressionType(precedence: 20,                        inline: .singleUseOnly)
```
Comment (saelo) [inline Sources/Fuzzilli/Lifting/JSExpressions.swift:18]: Maybe add a comment here or in the JavaScriptLifter.swift to explain why we handle them differently? I guess something like 
```swift 
// RegExp are objects, and so for example for the FuzzIL program 
//     v1 <- CreateRegExp
//     Compare v1, v1
// there is a difference between 
//    let a = /a/;
//    a === a;  // (true)
// and 
//     /a/ === /a/;  // (false)
// the former being the correct JavaScript equivalent.
```
or so?

File: Tests/FuzzilliTests/LifterTest.swift [hunk 2]
```diff
@@ -736,6 +736,30 @@ class LifterTests: XCTestCase {
 
         XCTAssertEqual(lifted_program,expected_program)
     }
+
+    func testRegExpInline(){
+        let fuzzer = makeMockFuzzer()
+        let b = fuzzer.makeBuilder()
+
+        let v0 = b.loadRegExp("a", RegExpFlags(rawValue:0))
```
Comment (saelo) [inline Tests/FuzzilliTests/LifterTest.swift:744]: Nit: here and below, I think you can just use `b.loadRegExp("a", RegExpFlags())`

File: Tests/FuzzilliTests/LifterTest.swift [hunk 3]
```diff
@@ -736,6 +736,30 @@ class LifterTests: XCTestCase {
...
+        b.compare(v0,v0, with: Comparator.equal);
```
Comment (saelo) [inline Tests/FuzzilliTests/LifterTest.swift:745]: Nit: I think you can drop `Comparator` here and below: `b.compare(v0,v0, with: .equal);`

**Comment by oicu0619** [comment]
Fixed that~

**Comment by saelo** [comment]
Thanks!

**Pull request merged**